cpp/lib/src/MARC.cc

```diff
@@ -1592,6 +1592,31 @@ bool Record::addSubfield(const Tag &field_tag, const char subfield_code, const s
     return true;
 }
 
+bool Record::addSubfieldCreateFieldUnique(const Tag &field_tag, const char subfield_code, const std::string &subfield_value) {
+    const auto &field(findTag(field_tag));
+    bool success = true;
+    if (field == fields_.end()) {
+        success = insertField(field_tag, std::string(1, subfield_code) + subfield_value);
+    }
+    if (success) {
+        if (not this->hasSubfieldWithValue(field_tag, subfield_code, subfield_value)) {
+            success = addSubfield(field_tag, subfield_code, subfield_value);
+        }
+    }
+    return success;
+}
+
+
+bool Record::hasSubfieldWithValue(const Tag &field_tag, const char subfield_code, const std::string &subfield_value) const {
+    auto field(findTag(field_tag));
+    while (field != fields_.cend()) {
```

Comment:
please also check if the tag is still the same (fields_ contains all fields, not only the one with the tag you are looking for).